apps/evm/src/components/DomainNameOrEllipseAddress/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useGetAddressDomainName } from 'clients/api';
+import { Icon, Tooltip } from 'components';
+import EllipseAddress, { type EllipseAddressProps } from 'components/EllipseAddress';
+import { useTranslation } from 'libs/translations';
+import { useChainId } from 'libs/wallet';
+import { ChainId } from 'types';
+import { cn, truncateAddress } from 'utilities';
+import type { Address } from 'viem';
+
+type DomainNameAddressProps = {
+  showProvider?: boolean;
+  showTooltip?: boolean;
+  shouldEllipseAddress?: boolean;
+} & EllipseAddressProps;
+
+export const DomainNameOrEllipseAddress: React.FC<DomainNameAddressProps> = ({
+  address,
+  className,
+  showProvider = true,
+  showTooltip = true,
+  shouldEllipseAddress = true,
+  ...ellipseAddressProps
+}) => {
+  const { t } = useTranslation();
+  const { chainId } = useChainId();
+  const { data: domainNames } = useGetAddressDomainName({
+    accountAddress: address as Address,
+  });
```

Comment:
`address` can be `undefined` here:
```suggestion
  const { data: domainNames } = useGetAddressDomainName({
    accountAddress: address as Address,
  }, {
    enabled: !!address
  });
```